src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/PeriodicShardSyncManager.java

```diff
@@ -66,6 +66,14 @@ public synchronized TaskResult start() {
         return new TaskResult(null);
     }
 
+    /**
+     * Runs ShardSync once, without scheduling further periodic ShardSyncs.
+     * @return TaskResult if successful, otherwise propagate nested exception
+     */
+    public synchronized TaskResult syncShardsOnce() {
+        return runShardSync();
```

Comment:
In runShardSync method, the PeriodicShardSyncManager will elect leaders and may not run shardSyncTask. This means shardSync is not guaranteed to run by this worker. But in the worker.Initialize stage, we need to guarantee the worker run shard sync if shouldInitiateLeaseSync() returns true. So this syncShardsOnce should not call runShardSync, instead, it should simply call metricsEmittingShardSyncTask

Reply:
That makes sense. But if any Worker can run `syncShardsOnce` on bootstrap, instead of just elected leaders, doesn't that leave us vulnerable to a large fleet of workers coming online at the same time and hammering the lease table?

Granted we have some sleeping in `shouldInitiateLeaseSync()` to minimize contention, but still I thought the point of leader election on bootstrap was to handle this case.

Reply:
Discussed offline - the jitter in `shouldInitiateLeaseSync()` is the "leader election" we are using here, since current leader election mechanism uses lease table to determine eligible leaders, and the lease table in this case is empty. Updated to call ShardSyncTask directly.